### ISSUE TITLE
Any plans to update this library to swift 4.0?

### ISSUE THREAD

**Issue by dileep017** [yourkarma/JWT#177]
Are there any plans to update this library to Swift 4.0?

**Comment by lolgear** [comment]
@dileep017 what do you mean?
It is an objective-c library with support even iOS 6. ( I hope so )

**Comment by dileep017** [comment]
I meant are you planning to write a swift code for this library??

**Comment by lolgear** [comment]
@dileep017 
It is a pure objective-c library.
It could be used in projects with swift 4, I suppose.

**Comment by lolgear** [comment]
@dileep017 
solved?

**Comment by dileep017** [comment]
@lolgear  i was talking about native Swift library. Yes you can bridge the objective c library and use it in swift.But i haven`t seen a single swift written library supports SH,RS and ES algorithms.
If you get what i mean, i would like to contribute to write this library in swift with your inputs.

**Comment by lolgear** [comment]
@dileep017 
well, interesting. 
How do you want to organize development of this library?

Should it be a part of current JWT library?
Should it be stored somewhere else? ( In this scenario I suggest to put in [jwt group](https://github.com/jwt) )

**Comment by dileep017** [comment]
Yes i would like to ask about the same, how could we organize the development of this library?
Do i get any help any more hands or are you expecting me to work alone?
And also i need to know a basic stuff like how to create the source code as library and convert it into a framework and pod.
Yes it can be a part of current jwt library and we can store it in jwt group as you said.

**Comment by lolgear** [comment]
@dileep017 I will think about all stuff at end of this week.

let me conclude questions about it:

1. How to organize development?
2. Alone or pair/team?
3. Distribution stuff for swift. ( Package manager, cocoa pods, Carthage )
4. Part of JWT or another project?

**Comment by dileep017** [comment]
@lolgear Sure will wait for your update.

**Comment by dileep017** [comment]
Have you given any thoughts on this?

**Comment by lolgear** [comment]
@dileep017 Hi!
I wait for a reply from inner developer of JWT.

Recently I asked him about opportunity to move this repo to group ( organization ) that I mentioned above.
It is a corner stone for now, because I don't have permissions to extend development team here.

1. How to organize development?
2. Alone or pair/team?
3. Distribution stuff for swift. ( Package manager, cocoa pods, Carthage )
4. Part of JWT or another project?

**How to organize development?**
It depends on reply from JWT developer.
It would be nice to move this repo to another group ( organization/team ) and to continue development. In this case it would be simple to start.

**Alone or pair/team?**
I could help you, but all stuff should be rewritten from sketch ( because this library has three big problems - legacy, backward API compatibility, old iOS version support ).
Even tests should be rewritten.
However, I have some thoughts in making this simple.

**Distribution stuff for swift. ( Package manager, cocoa pods, Carthage )**
Do not know about Carthage, but, of course, it would be nice to support it.
CocoaPods and Package manager should be.
Not sure, but I think that Package manager is a nice replacement for both.

**Part of JWT or another project?**
Preferably not. Again, it would be nice to host it in the group that I mentioned above.

**Some other thoughts.**
version 3.0, I think it would be the last version. Legacy is unbearable now and should be rewritten.

**Comment by dileep017** [comment]
@lolgear  Okay sounds like a plan. But I am happy to write it alone,but i just need a guidance regards legacy,backward API Compatability and old iOS version support.Along with the distribution stuff.

**Comment by lolgear** [comment]
@dileep017 
Ok, first of all, try to rewrite algorithm part ( HS only ) in Swift manner ( with further easy-to-extend to RS and possibly ES ).
As I remember, it doesn't use some legacy of apple OS part.
It will be JWA module.
and do not care about distribution for now, it is simple when everything is done.

**Comment by dj-neza** [comment]
Hi guys!
I'm just wondering about how this Swift library is coming along. There is still no alternative as far as I searched and I believe a lot of people are looking for something like this. I don't have much experience, but if I can help in some way let me know.

**Comment by dileep017** [comment]
@lolgear  Sure Dimitry.I couldnt concentrate more on this work last month and occupied with my work. I will start writing algorithm part for HS in Swift and will update you on the progress.
@dj-neza  Hey we are just starting up this.You can ask @lolgear  for further updates.
